### PR TITLE
rpc: add additional ban time fields to listbanned

### PR DESCRIPTION
**Pull request by barton2526**
> This PR adds a `ban_duration` and `time_remaining` field to the `listbanned` RPC command.

Ref: https://github.com/bitcoin/bitcoin/pull/21602

### CODE REVIEW

**Review by iFoggz — APPROVED** [review]
utACK

File: src/rpc/net.cpp [hunk 1]
```diff
@@ -260,15 +260,18 @@ UniValue listbanned(const UniValue& params, bool fHelp)
 
     banmap_t banMap;
     g_banman->GetBanned(banMap);
+    const int64_t current_time{GetTime()};
```
Comment (jamescowens) [inline src/rpc/net.cpp:263]: This is ok. I noticed Bitcoin had the argument that GetTime() is technically deprecated now, but this is not the place to change that out. We need to replace all of the GetTime() calls but that should be done in a separate PR.

**Review by jamescowens — APPROVED** [review]
utACK

File: src/rpc/net.cpp (hunk 1 above)
Comment (jamescowens) [reply src/rpc/net.cpp:263]: #2330 enables the replacement of legacy GetTime() calls.
